src/check/runner/configuration/GlobalParameters.ts

```diff
@@ -4,11 +4,21 @@ import { Parameters } from './Parameters';
 /** @internal */
 const globalParametersSymbol = Symbol.for('fast-check/GlobalParameters');
 
+export type GlobalPropertyHookFunction = () => void;
+export type GlobalAsyncPropertyHookFunction = (() => Promise<unknown>) | (() => void);
+type ExtraGlobalParameters = {
+  beforeEach?: GlobalPropertyHookFunction;
+  afterEach?: GlobalPropertyHookFunction;
+  asyncBeforeEach?: GlobalAsyncPropertyHookFunction;
+  asyncAfterEach?: GlobalAsyncPropertyHookFunction;
+};
+
 /**
  * Type describing the global overrides
  * @public
  */
-export type GlobalParameters = Pick<Parameters<unknown>, Exclude<keyof Parameters<unknown>, 'path' | 'examples'>>;
+export type GlobalParameters = Pick<Parameters<unknown>, Exclude<keyof Parameters<unknown>, 'path' | 'examples'>> &
+  ExtraGlobalParameters;
```

Comment:
Directly merge Extra* in the GlobalPaarmeters as follow:

```suggestion
export type GlobalParameters = Pick<Parameters<unknown>, Exclude<keyof Parameters<unknown>, 'path' | 'examples'>> &
  {
  beforeEach?: GlobalPropertyHookFunction;
  afterEach?: GlobalPropertyHookFunction;
  asyncBeforeEach?: GlobalAsyncPropertyHookFunction;
  asyncAfterEach?: GlobalAsyncPropertyHookFunction;
};
```

It would be better in terms of generated doc.